### ISSUE TITLE
Set rightOffset and scrollToPosition async as well as setVisibleRange

### ISSUE THREAD

**Issue by ProgrammingLife** [tradingview/lightweight-charts#406]
I need to set exactly bars count per visible view so I set it like this:

```
chart.timeScale().setVisibleRange({
	from: values[valuesCount - 50].time,
	to: values[valuesCount - 1].time,
});
```

All works fine, ok but then I need to set some offset at the right bound of my chart so I do following:
```
chart.timeScale().resetTimeScale();
chart.timeScale().scrollToPosition(2, false);
chart.applyOptions({ timeScale: { rightOffset: 10, borderColor: 'red', } });
```

and it doesn't work anymore after calling setVisibleRange! If I remove setVisibleRange then all those three lines work great. But how to get rightOffset + setVisibleRange? I need 50 bars per view + a little offset from the right bound.

**Comment by timocov** [comment]
Ha-ha, gotcha! 😂 Did you call all that stuff in one callstack, right?

`setVisibleRange` is kind of "async" operations, which is applied on the next animation frame before the chart is drawn. Maybe we need to set all operations with visible time range make "async".

**Comment by ProgrammingLife** [comment]
@timocov wow, that works. Thanks!!

**Comment by timocov** [comment]
The implementation depends on fixes for #335.
